client/src/components/chess-game/ChessGame.js

```diff
@@ -149,49 +156,52 @@ export default class ChessGame extends React.Component {
     }
   }
 
-  updateBoardState() {
+  updateBoardState = () => {
     const board = this.chess.board();
     const currentPlayer = this.chess.turn();
-    // eslint-disable-next-line react/no-unused-state
     this.setState({ board, currentPlayer });
   }
 
-  calcPossibleMoves(param) {
+  calcPossibleMoves = (param) => {
     const moves = this.chess.moves({ ...param, verbose: true });
     return moves.map(move => move.to);
   }
 
   render() {
     const {
-      playerColor,
-      isGameOver,
-      gameOverMessage,
-      playerOneTime,
-      playerTwoTime,
+      playerColor, isGameOver, playerOneTime, playerTwoTime, playerOneRating,
+      playerTwoRating, winner, chessBoardWidth, chessBoardHeight,
     } = this.state;
     let { board } = this.state;
     let { squares } = this;
-    if (playerColor === 'b') {
+    if (playerColor === BLACK_PLAYER) {
       const chessBoardData = blackPlayerBoard(squares, board);
       board = chessBoardData.newBoard;
       squares = chessBoardData.newSquares;
     }
     return (
       <>
+        <Timer time={playerTwoTime} />
         <div id="chess-game-container">
-          <Timer time={playerTwoTime} />
           <ChessBoard
+            childElementRef={this.chessBoarChildRef}
             calcPossibleMoves={this.calcPossibleMoves}
             movePiece={this.movePiece}
             squares={squares}
             board={board}
           />
-          <Timer time={playerOneTime} />
-          <div id="game-over-overlay" style={{ display: isGameOver ? 'block' : 'none' }}>
-            <p>{ gameOverMessage }</p>
-            <button type="button" onClick={this.startGame}>Play Again</button>
-          </div>
+          {isGameOver && (
+            <GameOverOverlay
+              playerOneRating={playerOneRating}
+              playerTwoRating={playerTwoRating}
+              winner={winner}
+              startGame={this.startGame}
+              width={chessBoardWidth}
```

Comment:
why not have width & height 100%?
Do we show anything else in end game beside overlay?

Reply:
To show the overlay, div has position absolute and with width and height 100%, it takes width and height of the viewport, not of the parent element.